src/components/external/PayByLink/PayByLinkCreation/components/PayByLinkCreationFormContainer/usePayByLinkFormData.ts

```diff
@@ -0,0 +1,137 @@
+import { TranslationKey } from '../../../../../../translations';
+import { useWizardForm } from '../../../../../../hooks/form/wizard/useWizardForm';
+import { PBLFormValues } from '../types';
+import { DeepPartial } from '../../../../../types';
+
+type UsePayByLinkFormDataProps = {
+    storeIds?: string[] | string;
+    defaultValues?: DeepPartial<PBLFormValues>;
+};
+
+export const usePayByLinkFormData = ({ storeIds, defaultValues }: UsePayByLinkFormDataProps) => {
+    const [selectedStore, setSelectedStore] = useState<string>('');
+    const { i18n } = useCoreContext();
+    const { getPayByLinkConfiguration, createPBLPaymentLink, getPayByLinkSettings, getPayByLinkStores } = useConfigContext().endpoints;
+
+    // Stores query
+    const storesQuery = useFetch({
+        fetchOptions: { enabled: !!getPayByLinkStores },
+        queryFn: useCallback(async () => {
+            return getPayByLinkStores?.(EMPTY_OBJECT, {});
+        }, [getPayByLinkStores]),
+    });
+
+    const storesSelectorItems = useMemo(() => {
+        const stores: PayByLinkStoreDTO[] = storesQuery.data?.data ?? [];
+        return stores
+            .filter(store => !storeIds || storeIds.includes(store.storeCode || ''))
+            .map(({ storeCode, storeId }) => ({
+                id: storeId || '',
+                name: storeCode || '',
+            }));
+    }, [storesQuery.data, storeIds]);
+
+    // Configuration query (depends on selected store)
+    const configurationQuery = useFetch({
+        fetchOptions: { enabled: !!getPayByLinkConfiguration && !!selectedStore },
+        queryFn: useCallback(async () => {
+            return getPayByLinkConfiguration?.(EMPTY_OBJECT, { path: { storeId: selectedStore } });
+        }, [getPayByLinkConfiguration, selectedStore]),
+    });
+
+    // Settings query (depends on selected store)
+    const settingsQuery = useFetch({
+        fetchOptions: { enabled: !!getPayByLinkSettings && !!selectedStore },
+        queryFn: useCallback(async () => {
+            return getPayByLinkSettings?.(EMPTY_OBJECT, { path: { storeId: selectedStore } });
+        }, [getPayByLinkSettings, selectedStore]),
+    });
+
+    const termsAndConditionsProvisioned = useMemo(() => {
+        return !!settingsQuery.data?.termsOfServiceUrl;
+    }, [settingsQuery.data?.termsOfServiceUrl]);
+
+    const getFieldConfig = useCallback(
+        (field: keyof PaymentLinkConfiguration) => {
+            return configurationQuery.data?.[field];
+        },
+        [configurationQuery.data]
+    );
+
+    // Form steps configuration
+    const formSteps = useMemo(() => {
+        const skipStoreStep = storesSelectorItems.length === 1 && termsAndConditionsProvisioned;
+        return getFormSteps({ i18n, getFieldConfig }).filter(step => !(step.id === 'store' && skipStoreStep));
+    }, [configurationQuery.data, getFieldConfig, i18n, storesSelectorItems, termsAndConditionsProvisioned]);
```

Comment:
To avoid rendering store selection step

```suggestion
    // Form steps configuration
    const formSteps = useMemo(() => {
        if(!storesSelectorItems.length || (storesSelectorItems.length === 1 && !settingsQuery.data)) return [];
        const skipStoreStep = settingsQuery.data && storesSelectorItems.length === 1 && termsAndConditionsProvisioned;
        return getFormSteps({ i18n, getFieldConfig }).filter(step => !(step.id === 'store' && skipStoreStep));
    }, [configurationQuery.data, getFieldConfig, i18n, storesSelectorItems, termsAndConditionsProvisioned, settingsQuery.data]);

```